packages/packages.go

```diff
@@ -297,11 +297,63 @@ type Filter struct {
 	Capabilities   []string
 	SpecMin        *semver.Version
 	SpecMax        *semver.Version
+	Discovery      *DiscoveryFilter
 
 	// Deprecated, release tags to be removed.
 	Experimental bool
 }
 
+type DiscoveryFilter struct {
+	Fields DiscoveryFilterFields
+}
+
+func NewDiscoveryFilter(filter string) (*DiscoveryFilter, error) {
+	filterType, args, found := strings.Cut(filter, ":")
```

Comment:
Currently we only have a type of discovery method, so I didn't add support for more.

If we had, a request using more than one would set the parameter multiple times, yes.

I can add support for this, though it wouldn't be really useful at the moment. What do you prefer?

Reply:
Do you think I should add support for `discovery` appearing multiple times?

So these two queries would be equivalent:
```
GET /search?discovery=fields:process.pid&discovery=fields:user.id
```
```
GET /search?discovery=fields:process.pid,user.id
```

This would also serve as base to add other methods. Though I don't think this would be very useful by now.

Reply:
Probably, it is not required to do it now. I guess Kibana could put all the fields required in one query parameter, WDYT?

Reply:
Yes, there would be no reason to separate in multiple parameters.